con-base/src/main/java/ba/codecentric/base/domain/Hotel.java

```diff
@@ -182,24 +182,24 @@ public Time getCheckIn() {
     }
 
     public void setCheckIn(String checkIn) {
-        this.checkIn = timeCheck(checkIn);
+        this.checkIn = parseTimeCheck(checkIn);
     }
 
     public Time getCheckOut() {
         return checkOut;
     }
 
     public void setCheckOut(String checkOut) {
-        this.checkOut = timeCheck(checkOut);
+        this.checkOut = parseTimeCheck(checkOut);
     }
 
-    private Time timeCheck(String string) {
-        final String SECONDS = ":00";
-        final String REGEX = ("^([0-1]\\d|2[0-3]):([0-5]\\d)$");
-        if (string != null) {
-            if (string.matches(REGEX)) {
-                return Time.valueOf(string + SECONDS);
-            } else return null;
-        } return null;
+    private Time parseTimeCheck(String time) {
+        final String EXTEND_SECONDS = ":00";
+        final String TIME_WITHOUT_SECOND = ("^([0-1]\\d|2[0-3]):([0-5]\\d)$");
+        if(time != null) {
```

Comment:
As I previously stated, throwing `IllegalArgumentException` would be more suitable than doing `null` check. Passing `null` and getting `null` doesn't say anything.